### PR TITLE
Board crud

### PR DESCRIPTION
**Pull request by andrei1904**

### CODE REVIEW

File: UDT.Business/Implementation/BoardService.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            
+            _dataContext.Board.Remove(board);
+            await _dataContext.SaveChangesAsync();
+            
+            return true;
+        }
+
+        public async Task<Board> UpdateAsync(Board newBoard)
+        {
+            var board = await _dataContext.Board.FindAsync(newBoard.Id);
```
Comment (ahoffman02) [inline UDT.Business/Implementation/BoardService.cs]: You should use FirstOrDefaultAsync to handle exception when there is no object with given id in DB.

File: UDT.Business/Implementation/BoardService.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        private readonly DataContext _dataContext;
+
+        public BoardService(DataContext dataContext)
+        {
+            _dataContext = dataContext;
+        }
+
+        public async Task<List<Board>> GetAllAsync()
+        {
+            return await _dataContext.Board.ToListAsync();
```
Comment (ahoffman02) [inline UDT.Business/Implementation/BoardService.cs]: Use AsAsyncEnumerable

File: UDT.Business/Interfaces/IBoardService.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using UDT.Model.Entities;
+
+namespace UDT.Business.Interfaces
+{
+    public interface IBoardService
+    {
+        Task<List<Board>> GetAllAsync();
```
Comment (ahoffman02) [inline UDT.Business/Interfaces/IBoardService.cs]: Use IAsyncEnumerable instead of task<list>

File: UniversityDeadlineTracker/Startup.cs [hunk 4]
```diff
@@ -25,6 +26,8 @@ public void ConfigureServices(IServiceCollection services)
             services.AddDbContext<DataContext>(options => options.UseSqlServer(Configuration.GetConnectionString("DbConnection")));
 
             services.AddControllersWithViews();
+            
+            services.AddTransient<IBoardService, BoardService>();
```
Comment (ahoffman02) [inline UniversityDeadlineTracker/Startup.cs]: I think is better to AddScoped to have the same instance on a single request

**Review by PacoPakkun — APPROVED** [review]

**Pull request merged**